Rubberduck.Core/UnitTesting/TestEngine.cs

```diff
@@ -89,66 +89,110 @@ public void Run(IEnumerable<TestMethod> tests)
 
         private void RunInternal(IEnumerable<TestMethod> tests)
         {
-            var testMethods = tests as IList<TestMethod> ?? tests.ToList();
-            if (!testMethods.Any())
+            if (_state.Status != ParserState.Ready)
             {
                 return;
             }
 
-            var modules = testMethods.GroupBy(test => test.Declaration.QualifiedName.QualifiedModuleName);
-            foreach (var module in modules)
+            _state.SuspendParser(this, () =>
```

Comment:
I think the entire lambda in the call should be its own method.

Reply:
Just to confirm I'm understanding - you're suggesting that we should do `_state.SuspendParser(this, () => RunInternal())`, with the body from the lambda defined within the `RunInternal()`?

Reply:
which can subsequently be shortened to a method group as ` _state.SuspendParser(this, RunInternal)`, yes